src/main/java/frc/robot/swerve/SwerveSubsystem.java

```diff
@@ -379,136 +392,204 @@ public Command stop() {
+    return isInTolerance(
+        target, AutoAim.TRANSLATION_TOLERANCE_METERS, AutoAim.ROTATION_TOLERANCE_RADIANS);
+  }
+
+  public boolean isInTolerance(
+      Pose2d target, double translationalToleranceMeters, double angularToleranceRadians) {
+    Transform2d diff = getPose().minus(target);
+    return MathUtil.isNear(0.0, Math.hypot(diff.getX(), diff.getY()), translationalToleranceMeters)
+        && MathUtil.isNear(
+            target.getRotation().getRadians(),
+            getPose().getRotation().getRadians(),
+            angularToleranceRadians);
+  }
+
+  public Command autoAimToL1(double vxModifier, double vyModifier) {
+    return translateToPose(
+        () -> {
+          Rectangle2d nearestLine = FieldUtils.L1Targets.getNearestLine(getPose());
+          return new Pose2d(
+              nearestLine.nearest(getPose().getTranslation()), nearestLine.getRotation());
+        },
+        () -> {
+          ChassisSpeeds speedsModifierRobotRelative =
+              ChassisSpeeds.fromFieldRelativeSpeeds(vxModifier, vyModifier, 0.0, getRotation());
+          // Kill all front-back requested velocity (because we want the robot to strafe)
+          speedsModifierRobotRelative.vxMetersPerSecond = 0.0;
+          return ChassisSpeeds.fromRobotRelativeSpeeds(speedsModifierRobotRelative, getRotation());
+        });
   }
 
   public boolean nearL1() {
-    // TODO
-    return true;
+    return isInAutoAimTolerance(
+        new Pose2d(
+            L1Targets.getNearestLine(getPose()).nearest(getPose().getTranslation()),
+            L1Targets.getNearestLine(getPose()).getRotation()));
   }
 
   public boolean isNearL1Reef() {
     // TODO
     // Not sure why this is different from near l1??
-    return true;
+    return L1Targets.getNearestLine(getPose()).getDistance(getPose().getTranslation()) > 0.3;
   }
 
   public boolean isNearReef(double toleranceMeters) {
-    // TODO
-    return true;
+    return getPose()
+            .getTranslation()
+            .minus(
+                DriverStation.getAlliance().orElse(Alliance.Blue) == Alliance.Blue
+                    ? AutoAim.BLUE_REEF_CENTER
+                    : AutoAim.RED_REEF_CENTER)
+            .getNorm()
+        < 3.25;
   }
 
   public Command autoAimToL23() {
-    // TODO
-    return Commands.none();
+    return translateToPose(() -> FieldUtils.CoralTargets.getClosestTargetL23(getPose()));
   }
 
   public boolean nearL23() {
-    // TODO
-    return true;
+    return isInAutoAimTolerance(FieldUtils.CoralTargets.getClosestTargetL23(getPose()));
   }
 
   public Command autoAimToL4() {
-    // TODO
-    return Commands.none();
+    return translateToPose(() -> FieldUtils.CoralTargets.getClosestTargetL4(getPose()));
   }
 
   public boolean nearL4() {
-    // TODO
-    return true;
+    return isInAutoAimTolerance(FieldUtils.CoralTargets.getClosestTargetL4(getPose()));
   }
 
   public Command autoAimToOffsetAlgae() {
```

Comment:
```suggestion
  public Command autoAimToOffsetAlgaePose() {
```
or location or something